### PR TITLE
Fix VPC configs

### PR DESCRIPTION
**Pull request by rtitle**
Our VPC configs were a bit wonky before, hopefully this cleans things up. Comments inline.

cc @eap, hopefully this makes sense..

Have you read [CONTRIBUTING.md](../CONTRIBUTING.md) lately? If not, do that first.

I, the developer opening this PR, do solemnly pinky swear that:

- [ ] I've documented my API changes in Swagger

In all cases:

- [ ] Get two thumbsworth of review and PO signoff if necessary
- [ ] Verify all tests go green
- [ ] Squash and merge; you can delete your branch after this
- [ ] Test this change deployed correctly and works on dev environment after deployment

### CODE REVIEW

File: leonardo-example.conf [hunk 1]
```diff
@@ -9,6 +9,23 @@ dataproc {
   leoGoogleProject = "GOOGLE_PROJECT"    # the name of the google project to use during cluster startup
                                          # *this is not the project the cluster will be created in
   clusterUrlBase = "https://YOUR_DOMAIN/notebooks" # the base url to access your cluster
+
```
Comment (rtitle) [inline leonardo-example.conf]: I moved `networkTag`, `firewallRuleName`, `vpcNetwork`, and `vpcSubnet` from `ProxyConfig` to `DataprocConfig`. They are used in Dataproc cluster setup, so didn't really make sense in `ProxyConfig`.

File: leonardo-example.conf [hunk 2]
```diff
@@ -45,15 +62,6 @@ mysql {
 proxy {
   # Should match the jupyter wildcard cert specified in command above
   jupyterDomain = "JUPYTER_DOMAIN_NAME"
-  # This tag is used to associate the cluster's master with Leonardo's firewall rule.
-  networkTag = "leonardo"
-  # The VPC network that the firewall rule is applied to. Most instances of leo should use
-  # 'default' since it is the VPC allocated in all newly configured GCP projects.
-  firewallVPCNetwork = "default"
```
Comment (rtitle) [inline leonardo-example.conf]: Previously `firewallVPCNetwork` was never used! I made it optional, and set it if it's present (our Leo instance leaves it unset).

File: src/main/resources/reference.conf [hunk 3]
```diff
@@ -4,6 +4,11 @@ dataproc {
   applicationName = "firecloud:leonardo"
   dataprocDockerImage = "gcr.io/broad-dsde-prod/leonardo-notebooks:prod"
   jupyterServerName = "jupyter-server"
+  firewallRuleName = "leonardo-notebooks-rule"
+  networkTag = "leonardo"
+  # Unset to use project-level defaults. Note a VPC/subnet with the literal name "default" may not exist.
+  #vpcNetwork = "default"
```
Comment (rtitle) [inline src/main/resources/reference.conf]: Should fix the issue in broad-dsde-dev: that project doesn't have a VPC/subnet named "default", but it works it you don't set it and let Google figure it out.

File: src/main/scala/org/broadinstitute/dsde/workbench/leonardo/dao/google/HttpGoogleDataprocDAO.scala [hunk 4]
```diff
@@ -190,10 +191,20 @@ class HttpGoogleDataprocDAO(appName: String,
   private def getClusterConfig(machineConfig: MachineConfig, initScript: GcsPath, clusterServiceAccount: Option[WorkbenchEmail], credentialsFileName: Option[String], stagingBucket: GcsBucketName): DataprocClusterConfig = {
     // Create a GceClusterConfig, which has the common config settings for resources of Google Compute Engine cluster instances,
     // applicable to all instances in the cluster.
-    // Set the network tag, network, and subnet. This allows the creaed GCE instances to be exposed by Leo's firewall rule.
-    val gceClusterConfig = new GceClusterConfig()
-      .setTags(List(defaultNetworkTag.value).asJava)
-      .setSubnetworkUri(defaultVPCSubnet.value)
+    // Set the network tag, network, and subnet. This allows the created GCE instances to be exposed by Leo's firewall rule.
+    val gceClusterConfig = {
```
Comment (eap) [inline src/main/scala/org/broadinstitute/dsde/workbench/leonardo/dao/google/HttpGoogleDataprocDAO.scala]: +1 
This is exactly what I would have preferred to do in my last PR but lacked the scala-foo to express coherently.

**Comment by rtitle** [comment]
@eap ok to merge this from your end? You'll probably have to change some configs but the functionality should would for both our Leo instances.

**Comment by eap** [comment]
Thanks @rtitle, ok to merge for us.

**Pull request merged**